lib/metadata/index.js

```diff
@@ -218,20 +233,36 @@ Metadata.prototype._refreshKeyspacesCb = function (waitReconnect, callback) {
   });
 };
 
+Metadata.prototype._getKeyspaceReplicas = function (keyspace) {
+  if (!keyspace.replicas) {
+    //Calculate replicas the first time for the keyspace
+    keyspace.replicas =
+      keyspace.tokenToReplica(this.tokenizer, this.ringTokensAsStrings, this.primaryReplicas, this.datacenters);
+  }
+  return keyspace.replicas;
+};
+
 /**
- * Gets the host list representing the replicas that contain such partition.
+ * Gets the host list representing the replicas that contain the given partition key, token or token range.
  * <p>
  *   It uses the pre-loaded keyspace metadata to retrieve the replicas for a token for a given keyspace.
  *   When the keyspace metadata has not been loaded, it returns null.
  * </p>
  * @param {String} keyspaceName
- * @param {Buffer} tokenBuffer
+ * @param {Buffer|Token|TokenRange} token Can be Buffer (serialized partition key), Token or TokenRange
  * @returns {Array}
  */
-Metadata.prototype.getReplicas = function (keyspaceName, tokenBuffer) {
+Metadata.prototype.getReplicas = function (keyspaceName, token) {
   if (!this.ring) {
     return null;
   }
+  if (util.isBuffer(token)) {
```

Comment:
Minor nit: `util.isBuffer()` is deprecated, use `Buffer.isBuffer()` instead or `token instanceof Buffer` (no difference really).

Reply:
ah, right!  I'll fix that